packages/lib/test/utils.test.ts

```diff
@@ -0,0 +1,69 @@
+import { hashEdwardsPublicKey, publicKeyFromString } from "../src/utils";
+import { EdwardsPoint, WeierstrassPoint } from "../src/babyJubjub";
+// @ts-ignore
+import { buildPoseidonReference } from "circomlibjs";
+
+describe("signature and key parsing utilities", () => {
+  test("should parse an encoded public key", () => {
+    const encodedPubKey =
+      "041052d6da0c3d7248e39e08912e2daa53c4e54cd9f2d96e3702fa15e77b199a501cd835bbddcc77134dc59dbbde2aa702183a68c90877906a31536eef972fac36";
+
+    const parsedPubKey = publicKeyFromString(encodedPubKey);
+
+    const expectedPubKey = new WeierstrassPoint(
```

Comment:
Where did this expectedPubKey come from? We should have documentation for how these were all computed (and ideally they weren't just from the utils function we're testing LOL).

Reply:
Will add documents to all the tests